etcdserver/api/v3lock/v3lockpb/v3lock.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package v3lockpb;
+
+import "gogoproto/gogo.proto";
+import "etcd/etcdserver/etcdserverpb/rpc.proto";
+
+// for grpc-gateway
+import "google/api/annotations.proto";
+
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+service Lock {
+  rpc Lock(LockRequest) returns (LockResponse) {
+      option (google.api.http) = {
+        post: "/v3alpha/lock/lock"
+        body: "*"
+    };
+  }
+  rpc Unlock(UnlockRequest) returns (UnlockResponse) {
```

Comment:
The key will only be returned on lock acquire. So long as `Key` exists with create rev <= response.header.revision (for most cases you'll only need the exists check), it holds the lock. That way it can interact with Txns safely.

I think it only needs some documentation, not a full RPC.